CI: Cilium L4LB XDP: clang: command not found

https://github.com/cilium/cilium/runs/3539873027?check_suite_focus=true
[logs_177932.zip](https://github.com/cilium/cilium/files/7127245/logs_177932.zip)

```
    default: ================================================================================
    default: Install  39 Packages
    default: Upgrade   5 Packages
    default: 
    default: Total download size: 153 M
    default: Downloading Packages:
    default: 
    default: The downloaded packages were saved in cache until the next successful transaction.
    default: You can remove cached packages by executing 'dnf clean packages'.
    default: Error: Error downloading packages:
    default:   Status code: 503 for https://mirrors.fedoraproject.org/metalink?repo=updates-released-f34&arch=x86_64 (IP: 140.211.169.196)
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'fedora/34-cloud-base' version '34.20210423.0' is up to date...
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: flag to force provisioning. Provisioners marked to run always will still run.
0s
0s
Run ssh default "sudo /bin/sh -c 'cd /vagrant/test/l4lb && ./test.sh cilium ad01ab6f73bd9bcf4d828a37a0cbbe3ce44a47d8'"
+ IMG_OWNER=cilium
+ IMG_TAG=ad01ab6f73bd9bcf4d828a37a0cbbe3ce44a47d8
+ clang -O2 -Wall -target bpf -c bpf_xdp_veth_host.c -o bpf_xdp_veth_host.o
./test.sh: line 17: clang: command not found
Error: Process completed with exit code 127.
```

It seems like maybe we failed during the VM provisioning and didn't detect it because the retry exited with a zero exit code.

@brb Maybe we should retry with `vagrant up --provision` instead of just `vagrant up`, to force provisioning?

--- a/.github/workflows/tests-l4lb-v1.10.yaml
+++ b/.github/workflows/tests-l4lb-v1.10.yaml
@@ -157,7 +157,7 @@ jobs:
           # Retry if it fails (download.fedoraproject.org returns 404 sometimes)
           # Spend up to 10 seconds on this
           for i in {1..4}; do
-            if vagrant up; then
+            if vagrant up --provision; then
               break
             fi
             sleep $i

--- a/.github/workflows/tests-l4lb.yaml
+++ b/.github/workflows/tests-l4lb.yaml
@@ -160,7 +160,7 @@ jobs:
           # Retry if it fails (download.fedoraproject.org returns 404 sometimes)
           # Spend up to 10 seconds on this
           for i in {1..4}; do
-            if vagrant up; then
+            if vagrant up --provision; then
               break
             fi
             sleep $i

